aten/src/ATen/native/TensorShape.cpp

```diff
@@ -545,6 +545,23 @@ std::vector<Tensor> tensor_split(const Tensor& self, IntArrayRef indices, int64_
   return splits;
 }
 
+std::vector<Tensor> tensor_split(const Tensor& self, const Tensor& tensor_indices_or_sections, int64_t dim) {
+  TORCH_CHECK(self.dim() > 0, "expected at least a 1-dimensional tensor");
+  TORCH_CHECK(tensor_indices_or_sections.device() == kCPU,
+    "expected tensor_indices_or_sections to be on cpu")
```

Comment:
"expected tensor_indices_or_sections to be on cpu" -> "tensor_split() expected tensor_indices_or_sections to be on cpu, but it's on DEVICE_TYPE"

Reply:
(where DEVICE_TYPE is populated with the current device of `tensor_indices_or_sections`)